wps/core/models.py

```diff
@@ -16,20 +16,92 @@
 """Defines dataclasses for business-logic data as well as request/reply models for use
 in the API."""
 
-from pydantic import BaseModel
+from dataclasses import dataclass
+from enum import Enum
 
+from ghga_service_commons.utils.utc_dates import DateTimeUTC
+from pydantic import BaseModel, EmailStr, Field
 
-class WorkPackageData(BaseModel):
+__all__ = [
+    "WorkType",
+    "WorkOrderToken",
+    "WorkPackageCreationData",
+    "WorkPackageData",
+    "WorkPackage",
+]
+
+
+class BaseDto(BaseModel):
+    """Base model preconfigured for use as Dto."""
+
+    class Config:  # pylint: disable=missing-class-docstring
+        extra = "forbid"
+        frozen = True
+
+
+class WorkType(str, Enum):
+    """The type of work that a work package describes."""
+
+    DOWNLOAD = "download"
+    UPLOAD = "upload"
+
+
+@dataclass(frozen=True)
+class WorkOrderToken:
+    """A class describing a work order token."""
+
+    type: WorkType
+    file_id: str
+    user_id: str
+    public_key: str
+    full_user_name: str
+    email: str
+
+
+class WorkPackageCreationData(BaseDto):
+    """
+    All data necessary to create a work package.
+    """
+
+    user_id: str = Field(default=..., title="ID od the user")
```

Comment:
The titles are not necessary because pydantic will automatically provide something similar based on the parameter name unless you want to give some more details but that is probably better done in the description.

Reply:
Ok, will look where it makes sense to simplify.